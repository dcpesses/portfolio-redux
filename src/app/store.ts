import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import todosReducer from '../features/todos/todosSlice';

const reducer = {
  todos: todosReducer
};
export const store = configureStore({ reducer });

export function getStoreWithState(preloadedState?: RootState) {
  return configureStore({ reducer, preloadedState });
}

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
