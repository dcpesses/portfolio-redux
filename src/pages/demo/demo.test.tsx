import {render, screen} from '@testing-library/react';
import {BrowserRouter} from 'react-router';
import {Provider} from 'react-redux';
import {store} from '@/app/store';

import Demo from './index';

describe('Demo', () => {
  test('Should render as expected', () => {
    const {container} = render(
      <Provider store={store}>
        <BrowserRouter>
          <Demo />
        </BrowserRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  test('Should render with Home link', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Demo />
        </BrowserRouter>
      </Provider>
    );

    expect(screen.getByText(/Home/i)).toBeInTheDocument();
  });
});
