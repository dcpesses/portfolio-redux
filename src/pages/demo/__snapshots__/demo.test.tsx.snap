// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Demo > Should render as expected 1`] = `
<div>
  <div
    class="Demo"
  >
    <header
      class="Demo-header"
    >
      <div
        style="css-float: left;"
      >
        <a
          class="Demo-link"
          data-discover="true"
          href="/"
        >
          ← Home
        </a>
      </div>
      <div
        style="css-float: right;"
      >
        Edit 
        <code>
          pages/demo/index.tsx
        </code>
         and save to test HMR updates.
      </div>
    </header>
    <h1>
      Demos & Examples
    </h1>
    <div
      class="Demo-examples"
    >
      <div>
        <div
          class="Demo-example"
        >
          <h3
            class="Demo-subheader"
          >
            Redux To-Dos
          </h3>
          <div
            class="todos"
          >
            <div
              class="add-todo"
            >
              <input
                placeholder="Enter a Todo..."
                type="text"
                value=""
              />
              <button>
                Add Todo
              </button>
            </div>
            <ul
              class="todos-list"
            >
              <li
                class="todo-list-item"
              >
                <span
                  class="item-checkbox"
                >
                  <input
                    title="Mark Complete"
                    type="checkbox"
                  />
                </span>
                <span
                  class="item-content"
                  style="text-decoration: none;"
                >
                  Hit the gym
                </span>
                <span
                  class="item-actions"
                >
                  <button
                    class="item-actions-edit"
                    title="Edit"
                  >
                    ✎
                  </button>
                  <button
                    class="item-actions-delete"
                    title="Delete"
                  >
                    ✖
                  </button>
                </span>
              </li>
              <li
                class="todo-list-item"
              >
                <span
                  class="item-checkbox"
                >
                  <input
                    checked=""
                    title="Mark Incomplete"
                    type="checkbox"
                  />
                </span>
                <span
                  class="item-content"
                  style="text-decoration: line-through;"
                >
                  Meet George
                </span>
                <span
                  class="item-actions"
                >
                  <button
                    class="item-actions-edit"
                    title="Edit"
                  >
                    ✎
                  </button>
                  <button
                    class="item-actions-delete"
                    title="Delete"
                  >
                    ✖
                  </button>
                </span>
              </li>
            </ul>
          </div>
          <div />
        </div>
        <div
          class="Demo-example"
        >
          <h3
            class="Demo-subheader"
          >
            Redux Counter
          </h3>
          <div>
            <div
              class="_row_ca4d94"
            >
              <button
                aria-label="Decrement value"
                class="_button_ca4d94"
              >
                 - 
              </button>
              <span
                aria-label="Counter value"
                class="_value_ca4d94"
              >
                0
              </span>
              <button
                aria-label="Increment value"
                class="_button_ca4d94"
              >
                 + 
              </button>
            </div>
            <div
              class="_row_ca4d94"
            >
              <input
                aria-label="Set increment amount"
                class="_textbox_ca4d94"
                value="2"
              />
              <button
                aria-label="Add increment amount"
                class="_button_ca4d94"
              >
                Add
              </button>
              <div>
                <button
                  aria-label="Add async increment amount"
                  class="_asyncButton_ca4d94"
                >
                  Add Async
                </button>
                <button
                  aria-label="Add increment amount if odd"
                  class="_button_ca4d94"
                >
                  Add If Odd
                </button>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="Demo-example"
        >
          <h3
            class="Demo-subheader"
          >
            State Counter
          </h3>
          <div>
            <div
              class="_row_1843ce"
            >
              <button
                aria-label="Decrement value"
                class="_button_1843ce"
              >
                 - 
              </button>
              <span
                aria-label="Counter value"
                class="_value_1843ce"
              >
                0
              </span>
              <button
                aria-label="Increment value"
                class="_button_1843ce"
              >
                 + 
              </button>
            </div>
            <div
              class="_row_1843ce"
            >
              <input
                aria-label="Set increment amount"
                class="_textbox_1843ce"
                value="2"
              />
              <button
                aria-label="Add increment amount"
                class="_button_1843ce"
              >
                Add
              </button>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
